src/SportsHub.Web/Controllers/TeamsController.cs

```diff
@@ -87,5 +94,42 @@ public async Task<IActionResult> EditTeam([FromForm] EditTeamModel editTeamModel
 
             return Ok();
         }
+
+        [HttpGet("{location}/{categoryName}/{subcategoryName}")]
+        /*[Authorize(Policies.Admin)]*/
+        [AllowAnonymous]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+
+        public async Task<IActionResult> GetFilteredTeams(string location, string categoryName, string subcategoryName)
+        {
+            var teams = await _teamService.GetSortedTeamAsync();
+
+            if (categoryName != "All")
```

Comment:
We shouldn't bind on the name of the model in this case. What would happen if I named the category "All"? 
```suggestion
            if (string.IsNullOrEmpty(categoryName))
```